src/System.Net.NameResolution/src/System.Net.NameResolution.csproj

```diff
@@ -216,6 +222,11 @@
     </Compile>
   </ItemGroup>
 
+  <ItemGroup Condition="'$(TargetGroup)' == 'net46'">
```

Comment:
I don't think it's really obvious what net46 vs netcore50 actually mean for an external developer or even to internal developers that don't work much with contracts and generation names.
I think we should add comments describing what each ItemGroup is meant for. (E.g. "Support for .Net Desktop", "Support for DNX on Windows", etc.)